### PR TITLE
Update README.md to link to html doc and correct api compile error

### PR DESCRIPTION
**Pull request by mcarbonneaux**

### CODE REVIEW

File: conf2sql.pl [hunk 1]
```diff
@@ -29,16 +29,23 @@
 
 die "$program: missing --dbdriver option\n" unless defined($opts{'dbdriver'});
 die "$program: missing --dbname option\n" unless defined($opts{'dbname'});
+if (!$opts{'dbdriver'} =~ /sqlite/i) {
```
Comment (Castaglia) [inline conf2sql.pl:32]: Why is this condition needed?

File: sqlite-conf.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
+DROP TABLE ftpctxt;
```
Comment (Castaglia) [inline sqlite-conf.sql]: Ideally the document changes would be in a separate PR from the schema-related changes, as they don't relate to each other.

File: conf2sql.pl (hunk 1 above)
Comment (mcarbonneaux) [inline conf2sql.pl:32]: sqlite note user server/pass/user like mysql and pgsql, is to differentiate, and i use dbname as sqlite dsn.

Comment (mcarbonneaux) [inline conf2sql.pl:32]: i'm testing your module with sqlite, and i'm correcting each problem i found.
the first was in conf2sql.pl to support sqlite!
this modification make possible tu use sqlite with this script.
the use is: ./conf2sql.pl --dbdriver=SQLite --dbname=/path/to/sqlite.db

File: sqlite-conf.sql (hunk 2 above)
Comment (mcarbonneaux) [inline sqlite-conf.sql]: this is the sqlite version mysql-conf.sql and postgres.sql file

File: mod_conf_sql.c [hunk 3]
```diff
@@ -157,6 +157,12 @@ static int sqlconf_parse_uri_db(char **uri) {
   *tmp = '\0';
   sqlconf_db.database = pstrdup(sqlconf_pool, *uri);
 
+  int i;
+  int n=strlen(sqlconf_db.server);
+
+  // replace \ with /
+  for(i=0;i<=n;i++) if(sqlconf_db.server[i]=='\\') sqlconf_db.server[i]='/';
```
Comment (Castaglia) [inline mod_conf_sql.c]: Could you provide an example of where this substitution is necessary?

File: README.md [hunk 4]
```diff
@@ -4,4 +4,4 @@ proftpd-mod_conf_sql
 The mod_conf_sql module for ProFTPD allows for reading of configuration data
 (normally provided by proftpd.conf) from SQL tables.
 
-For additional information, see mod_conf_sql.html.
```
Comment (Castaglia) [inline README.md:7]: This portion of the PR has now been updated in the master branch.  Thanks!

File: conf2sql.pl (hunk 1 above)
Comment (Castaglia) [inline conf2sql.pl:32]: OK.  Please create a separate pull request for each issue found.  That makes it easier to review the pull request, and to modify/merge just the ones that work, while the others are refined.  I cannot and will not accept a single pull request which jumbles together multiple unrelated changes.

File: sqlite-conf.sql [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+CREATE TABLE ftpctxt (
+    id INTEGER PRIMARY KEY AUTOINCREMENT,
+    parent_id INTEGER UNSIGNED,
+    name VARCHAR(255),
+    type VARCHAR(255),
+    info VARCHAR(255)
+);
+
+DROP TABLE ftpconf;
+CREATE TABLE ftpconf (
```
Comment (Castaglia) [inline sqlite-conf.sql]: Why not use `CREATE TABLE ... IF NOT EXISTS`, instead of using `DROP TABLE ...` beforehand?

File: mod_conf_sql.c [hunk 6]
```diff
@@ -633,10 +639,10 @@ static modret_t *sqlconf_dispatch(cmd_rec *cmd, char *name) {
   if (!cmdtab) {
     pr_log_debug(DEBUG0, MOD_CONF_SQL_VERSION
       ": unable to find SQL hook symbol '%s'", name);
-    return ERROR(cmd);
+    return PR_ERROR(cmd);
   }
 
-  res = call_module(cmdtab->m, cmdtab->handler, cmd);
+  res = pr_module_call(cmdtab->m, cmdtab->handler, cmd);
```
Comment (Castaglia) [inline mod_conf_sql.c:646]: This has now been fixed in master.  Thanks!

File: sqlite-conf.sql (hunk 5 above)
Comment (mcarbonneaux) [inline sqlite-conf.sql]: yes good sugestion!
modified on mysql and postgresql...

File: mod_conf_sql.c (hunk 3 above)
Comment (mcarbonneaux) [inline mod_conf_sql.c]: when start proftpd with sql:// uri... and if you transmit absolute path to the sqlite file with / in file path... you must use another character than / in the path...

exemple :
proftpd -c sql://nouser:nopass@\path\to\sqlitefile.db/db:SQlite/ctxt:ftpctxt/conf:ftpconf/map:ftpmap

Comment (mcarbonneaux) [inline mod_conf_sql.c]: idealy the url while be : 
sql://(/path/to/dabasefile.db|databasename)?dbpass=pass&dbuser=user&dbhost=hostname&dbctxt=ftpctxt&conf=ftpconf&map=ftpmap

exemple for sqlite:
sql:///path/to/sqlitefile.db?dbctxt=ftpctxt&conf=ftpconf&map=ftpmap
exemple for mysql:
sql://databasename?dbuser=user&dbpass=pass&dbhost=hostname&dbctxt=ftpctxt&conf=ftpconf&map=ftpmap

Comment (Castaglia) [inline mod_conf_sql.c]: You might be running into shell handling of special characters.  That is, what if you enclosed the entire `-c` argument in single quotes, _e.g._:

```
$ proftpd -c 'sql://nouser:nopass@/path/to/sqlitefile.db/db:SQlite/ctxt:ftpctxt/conf:ftpconf/map:ftpmap'
```
